src/components/auth/Signup.jsx

```diff
@@ -1,7 +1,41 @@
+import { useReducer, useState } from "react";
 import auth from "./auth.module.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { postUserSignup } from "api.js";
+import { Error } from "../Error";
 
 export function Signup() {
+  const [isPwdVisible, setIsPwdVisible] = useState(false);
+  const [isConfirmPwdVisible, setIsConfirmPwdVisible] = useState(false);
+  const [error, setError] = useState("");
+  const navigate = useNavigate();
+
+  const [signupInfo, signupDispatch] = useReducer(signup, {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPwd: "",
+    tAC: false,
+  });
+
+  function signup(state, { type, payload }) {
+    switch (type) {
+      case "FIRSTNAME":
+        return { ...state, firstName: payload };
+      case "LASTNAME":
+        return { ...state, lastName: payload };
+      case "EMAIL":
+        return { ...state, email: payload };
+      case "PWD":
+        return { ...state, password: payload };
+      case "CONFIRMPWD":
+        return { ...state, confirmPwd: payload };
+      case "TAC":
+        return { ...state, tAc: !state.tAc };
+    }
```

Comment:
what TAC stands for?

Reply:
terms and conditions, i must have missed it will add it thanks for pointing out.